### PR TITLE
[AzureStack][AKS]Add AKS into AzureStack profile

### PR DESCRIPTION
**Pull request by hhz1992**
**Description**  
This PR adds Azure Kubernetes Service (AKS) and Azure Container Registry (ACR) CLI into Azure Stack profile given the AKS and ACR will be private preview around July 2020.
1. Add MGMT_CONTAINERSERVICE and MGMT_CONTAINERREGISTRY in 2019-03-01-hybrid profile
2. Update acs commands to have multi-api support

**Testing Guide**  

1. Switch to Azure Stack profile
2. Use AKS and ACR CLI to test

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acs/_client_factory.py [hunk 1]
```diff
@@ -59,8 +59,9 @@ def get_auth_management_client(cli_ctx, scope=None, **_):
 
 def get_container_service_client(cli_ctx, **_):
     from azure.mgmt.containerservice import ContainerServiceClient
+    from azure.cli.core.profiles import ResourceType, get_api_version
 
-    return get_mgmt_service_client(cli_ctx, ContainerServiceClient)
+    return get_mgmt_service_client(cli_ctx, ContainerServiceClient, api_version=get_api_version(cli_ctx, ResourceType.MGMT_CONTAINERSERVICE))
```
Comment (bganapa) [inline src/azure-cli/azure/cli/command_modules/acs/_client_factory.py:64]: formatting, i believe CI pyblinter will flag this

Comment (hhz1992) [reply src/azure-cli/azure/cli/command_modules/acs/_client_factory.py]: Fixed

**Comment by yungezz** [comment]
hi @arrownj, could you pls review the PR? thanks

**Comment by bganapa** [comment]
@arrownj please note that this is getting in to a side branch.
@Juliehzl  could you please help review? This is for AzureStack, adding AKS in to the existing profile.  We are updating the side branch as we are looking for a private preview

File: src/azure-cli/azure/cli/command_modules/acs/commands.py [hunk 2]
```diff
@@ -24,14 +24,16 @@
 def load_command_table(self, _):
 
     container_services_sdk = CliCommandType(
-        operations_tmpl='azure.mgmt.containerservice.v2017_07_01.operations.'
```
Comment (hhz1992) [inline src/azure-cli/azure/cli/command_modules/acs/commands.py:27]: @arrownj This PR basically aims to add AKS(CONTAINERSERVICE) CLI into AzureStack profile using a specific API version. Previous CONTAINERSERVICE python SDK version is hardcoded here. I followed the same pattern as other services, to remove the version and add operation_group allowing multi-api support. However, the test didn't pass, and throw error: {'load_commands': ModuleNotFoundError("No module named 'azure.mgmt.containerservice.operations'")}

I am not sure how to fix it, can you please help to provide some guidelines?

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 3]
```diff
@@ -220,7 +224,11 @@ def default_api_version(self):
         # API versions
         ResourceType.MGMT_APPSERVICE: '2018-02-01',
         ResourceType.MGMT_EVENTHUB: '2017-04-01',
-        ResourceType.MGMT_IOTHUB: '2019-03-22'
+        ResourceType.MGMT_IOTHUB: '2019-03-22',
+        ResourceType.MGMT_CONTAINERSERVICE: SDKProfile('2020-03-01', {
```
Comment (bganapa) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:228]: So we support 2020-03-01 api version?

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 4]
```diff
@@ -54,13 +54,13 @@ class ResourceType(Enum):  # pylint: disable=too-few-public-methods
     MGMT_APPSERVICE = ('azure.mgmt.web', 'WebSiteManagementClient')
     MGMT_IOTHUB = ('azure.mgmt.iothub', 'IotHubClient')
     MGMT_ARO = ('azure.mgmt.redhatopenshift', 'AzureRedHatOpenShiftClient')
+    MGMT_CONTAINERSERVICE = ('azure.mgmt.containerservice', 'ContainerServiceClient')
```
Comment (rjaini) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:57]: As discussed can we add hybrid profile for CONTAINERREGISTRY version 2019-07-01 as well.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 3 above)
Comment (hhz1992) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:228]: Yes

File: src/azure-cli/azure/cli/command_modules/acs/_client_factory.py [hunk 5]
```diff
@@ -58,9 +58,7 @@ def get_auth_management_client(cli_ctx, scope=None, **_):
 
 
 def get_container_service_client(cli_ctx, **_):
-    from azure.mgmt.containerservice import ContainerServiceClient
-
-    return get_mgmt_service_client(cli_ctx, ContainerServiceClient)
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_CONTAINERSERVICE)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acs/_client_factory.py:61]: Please avoid positional argument here for resource type info.

File: src/azure-cli/azure/cli/command_modules/acs/commands.py [hunk 6]
```diff
@@ -24,14 +24,16 @@
 def load_command_table(self, _):
 
     container_services_sdk = CliCommandType(
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/acs/commands.py:26]: Please add resource type information in CliCommandType.

Comment (bganapa) [reply src/azure-cli/azure/cli/command_modules/acs/commands.py:26]: @hhz1992 to address this

**Comment by Juliehzl** [comment]
@hhz1992 any update?

File: src/azure-cli/azure/cli/command_modules/acs/_client_factory.py (hunk 5 above)
Comment (hhz1992) [reply src/azure-cli/azure/cli/command_modules/acs/_client_factory.py:61]: @Juliehzl Could you please provide further guidelines here? I followed the same pattern as other services in the files like:

def cf_resources(cli_ctx, subscription_id=None):
    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
                                   subscription_id=subscription_id).resources


def cf_container_registry_service(cli_ctx, subscription_id=None):
    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_CONTAINERREGISTRY,
                                   subscription_id=subscription_id)

**Pull request merged**